tiller_python_service/error_handlers.py

```diff
@@ -5,6 +5,29 @@
 
 LOGGER = logging.getLogger(__name__)
 
+def my_new_sorting_function(arr):
+    n = len(arr)
+    # optimize code, so if the array is already sorted, it doesn't need
+    # to go through the entire process
+    swapped = False
```

Comment:
This code has a lot of nesting, can we replace it with something that is broken out or use comprehension and functional patterns? help tiller